mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveMailAttributeTest.java

```diff
@@ -19,69 +19,84 @@
+        assertThat(removeMailet.getMailetInfo()).isEqualTo("Remove Mail Attribute Mailet");
+    }
 
-        mailet.init(mci);
-        return mailet;
+    @Test(expected = MailetException.class)
+    public void initShouldThrowExceptionIfMailetConfigDoesNotContainAttribute() throws MessagingException {
+        removeMailet.init(mailetConfig);
     }
 
+    @Test(expected = NullPointerException.class)
+    public void serviceShouldThrowExceptionWithMailNull() throws MessagingException {
+        mailetConfig.setProperty("name", "attribute1, attribute2");
+        removeMailet.init(mailetConfig);
 
-    @Test
-    public void testRemoveMailAttribute() throws MessagingException {
-        Mail m = createMail();
-        Mailet mailet = setupMailet(MAIL_ATTRIBUTE_NAME1);
+        removeMailet.service(mockMail);
+    }
 
-        // check if the mail has a attribute
-        Assert.assertNotNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME1));
-        Assert.assertNotNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME2));
+    @Test
+    public void serviceShouldDoNothingIfMailHasEmptyAttribute() throws MessagingException {
+        mailetConfig.setProperty("name", "attribute1, attribute2");
+        removeMailet.init(mailetConfig);
 
-        mailet.service(m);
+        mockMail = FakeMail.builder().build();
+        removeMailet.service(mockMail);
 
-        // Check if the attribute was removed
-        Assert.assertNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME1));
-        Assert.assertNotNull("Attribute deleted", m.getAttribute(MAIL_ATTRIBUTE_NAME2));
+        assertThat(mockMail.getAttributeNames()).isEmpty();
     }
 
+    @Test
+    public void serviceShouldDoNothingIfMailHasNotMatchAttribute() throws MessagingException {
+        mailetConfig.setProperty("name", "attribute1, attribute2");
+        removeMailet.init(mailetConfig);
+
+        mockMail = FakeMail.builder().attribute("attribute3", "anyValue").build();
+        removeMailet.service(mockMail);
+
+        assertThat(mockMail.getAttributeNames()).containsExactly("attribute3");
+    }
 
     @Test
-    public void testInvalidConfig() throws MessagingException {
-        boolean exception = false;
-        try {
-            setupMailet(null);
-        } catch (MessagingException e) {
-            exception = true;
-        }
-
-        Assert.assertTrue("invalid Config", exception);
+    public void serviceShouldRemoveSepecifiedAtribute() throws MessagingException {
+        mailetConfig.setProperty("name", "attribute1, attribute2");
+        removeMailet.init(mailetConfig);
+
+        mockMail = FakeMail.builder()
+            .attributes(ImmutableMap.<String, Serializable>of("attribute1", "value1", "attribute2", "value2", "attribute3", "value3"))
+            .build();
+        removeMailet.service(mockMail);
+
+        assertThat(mockMail.getAttributeNames()).containsExactly("attribute3");
     }
```

Comment:
> I follow the idea that the value on testing method should be there so it's easier to read

So that if I modify it, I should modify it in several place ?

You can enforce readability by naming : ATTRIBUTE_1, ATTRIBUTE_2, ATTRIBUTE_3, VALUE_1, VALUE_2, VALUE_3, hence you get readability, and your test is easier to change.